src/hotspot/share/gc/g1/g1ConcurrentMark.cpp

```diff
@@ -3052,7 +3052,7 @@ bool G1PrintRegionLivenessInfoClosure::do_heap_region(G1HeapRegion* r) {
   size_t remset_bytes    = r->rem_set()->mem_size();
   size_t code_roots_bytes = r->rem_set()->code_roots_mem_size();
   const char* remset_type = r->rem_set()->get_short_state_str();
-  uint cset_group_id     = 0;
+  uint cset_group_id     = G1CSetCandidateGroup::NoRemSetId;
```

Comment:
I suggest rewriting the following lines to sth like:

```c
uint id;
if (has_remset) {
  id = 
} else {
  id = no_remset_id;
}
```

Reply:
I prefer initializing the variable instead of a redundant `else`, but ymmv.